py/desispec/resolution.py

```diff
@@ -94,4 +105,32 @@ def to_fits_array(self):
         """
         return self.data
 
-#- (Unit tests moved to desispec.test.test_resolution)
\ No newline at end of file
+def _gauss_pix(x, mean=0.0, sigma=1.0):
+    """
+    Utility function to integrate Gaussian density within pixels
+
+    Args:
+        x (1D array): pixel centers
```

Comment:
You mention here that the centers must be on a grid but, even better, would be to remove this restriction so, for example, this code would work with BOSS spectra where the new constructor would actually be quite useful for interfacing with the DESI pipeline.  Your error function calculation is already general enough, you just need to pass in the bin egdes here instead of bin centers.